ui/desktop/src/components/settings_v2/providers/modal/ProviderConfiguationModal.tsx

```diff
@@ -62,28 +73,87 @@ export default function ProviderConfigurationModal() {
     closeModal();
   };
 
+  const handleDelete = () => {
+    setShowDeleteConfirmation(true);
+  };
+
+  const handleConfirmDelete = async () => {
+    try {
+      // Remove the provider configuration
+      // get the keys
+      const params = currentProvider.metadata.config_keys;
+
+      console.log('trying to delete', params);
+
+      // go through the keys are remove them
+      for (const param of params) {
+        console.log('param', param.name, 'secret', param.secret);
+        await remove(param.name, param.secret);
+      }
+
+      // Call onDelete callback if provided
+      // This should trigger the refreshProviders function
+      if (modalProps.onDelete) {
+        modalProps.onDelete(currentProvider.name);
+      }
+
+      // Close the modal after deletion and callback
+      closeModal();
+    } catch (error) {
+      console.error('Failed to delete provider:', error);
+      // Keep modal open if there's an error
+    }
+  };
+
+  const handleCancelDelete = () => {
```

Comment:
I would just inline `setShowDeleteConfirmation` below in an anonymous function to avoid having a wrapper function with a different name. Just for readability/simplicity.

ie...

```
onCancelDelete={() => setShowDeleteConfirmation(false)}
```